File: Patterns/Pattern21.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

const pattern = (n) => {
    console.log("*".repeat(n));
    
    // Middle rows
    for (let i = 0; i < n - 2; i++) {
        console.log("*" + " ".repeat(n - 2) + "*");
    }
    
    // Last row
    console.log("*".repeat(n));
}

rl.question('Enter something: ', (n) => {
    pattern(n);
  rl.close();
});

File: Patterns/Pattern21.cpp
#include <iostream>

using namespace std;

void pattern(int n){
    // for(int i=0;i<n;i++){
    //     cout << "*" ;
        
    // }
    // cout << endl;
    // for(int i=0;i<n-2;i++){
    //     for (int j = 0; j < 1; j++)
    //     {
    //         cout << "*" ;
    //     }
    //     for (int j = 0; j < n-2; j++)
    //     {
    //         cout << " " ;
    //     }
    //     for (int j = 0; j < 1; j++)
    //     {
    //         cout << "*" ;
    //     }
        
    //     cout << endl;
    // }
    // for(int i=0;i<n;i++){
    //     cout << "*" ;
        
    // }
     cout << string(n, '*') << endl;

    // Middle rows
    string middleRow = "*" + string(n - 2, ' ') + "*";
    for (int i = 0; i < n - 2; i++) {
        cout << middleRow << endl;
    }

    // Last row
    cout << string(n, '*') << endl;
}

// Input: ‘N’ = 4
// Output: 
// ****
// *  *
// *  *
// ****
// Input: ‘N’ = 5
// Output: 
// *****
// *   *
// *   *
// *   *
// *****


int main()
{   int n;
    cout << " Enter Number here : " ; cin >> n;
    pattern(n);
    return 0;
}